library/src/main/java/kwasm/ast/Memory.kt

```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package kwasm.ast
+
+import kwasm.format.text.Type
+
+/** This data class is used to represent the structure of a [Type.MemoryType] */
+data class Memory(val limits: Type.Limits)
```

Comment:
Actually wait a minute. I shouldn't do this. The ast should be independent of the format of WASM we're reading in. I gotta think how I want to reconfigure this.

Reply:
There should be a Limits ast value too.

Reply:
Done